### ISSUE TITLE
Discrepency in pi estimates between vcftools and pixy

### ISSUE THREAD

**Issue by nitinra** [ksamuk/pixy#14]
Hello, 

I calculated pi diversity using both vcftools and pixy. The estimates from vcftools range from 0-0.0020 whereas the ones from pixy range from 0-0.4. Here is the pipleline I used to generate these. 

Generated multi sample vcf file using haplotype caller, genomicsdbimport and genotypevcfs (I did not include the `-all-sites` option while running genotypegvcfs.

VCFfile pre-filteration:

`vcftools --vcf input.vcf --max-missing 0.1 --minQ 30 --maf 0.1 --remove lowdepthindividuals --recode --recode-INFO-all --out output_filtered.vcf`
`bcftools +prune -l 0.2 -w 50kb output_filtered.vcf -Ov -o output_filtered_ldpruned.vcf`

Pi calculations using VCFtools:
`vcftools --vcf output_filtered_ldpruned.vcf --window-pi 10000 --out pi`

Pi calculations using pixy:

`pixy --stats pi --vcf output_filtered_ldpruned.vcf --zarr_path ./zarr \
--window_size 10000 --populations allpop.list --bypass_filtration yes \
    --bypass-invariant-sites yes --outfile_prefix results/combined`

Here are the resulting files:
From pixy:
<img width="929" alt="Screen Shot 2020-12-29 at 9 53 47 AM" src="https://user-images.githubusercontent.com/70153351/103305378-f8bf9200-49d8-11eb-9f6d-51292a53e93c.png">

From vcftools:
<img width="933" alt="Screen Shot 2020-12-29 at 9 53 32 AM" src="https://user-images.githubusercontent.com/70153351/103305395-037a2700-49d9-11eb-89b4-5a10c8d9859a.png">

What could be causing such a drastic difference in pi estimates? Should I rerun pixy by including all sites again?

Thank you!

**Comment by ksamuk** [comment]
Hi There! This is the expected behavior of pixy. If you have a look at our preprint, you will see that VCFtools makes the incorrect assumption that missing sites are invariant. Pixy does not make this assumption, but needs a VCF with invariant sites in order to work correctly. The bypass invariants flag throws a major warning that is meant to alert you to fact that you  are bypassing this core functionality and will likely get incorrect estimates. That is the source of the discrepancy. 

So, in short, in order to get correct estimates, you'll need to generate a new VCF with GATK that includes invariant sites.

**Comment by nitinra** [comment]
Thank you so much for your explanation! I shall try it running again with a vcf file with all invariant sites.
